src/poetry/console/commands/self/source/add.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+from poetry.console.commands.self.self_command import SelfCommand
+from poetry.console.commands.source.add import SourceAddCommand
+
+
+class SelfSourceAddCommand(SelfCommand, SourceAddCommand):
+    name = "self source add"
+    description = "Add additional sources to Poetry's runtime environment."
+    options = [
+        o for o in SourceAddCommand.options if o.name in {"default", "secondary"}
```

Comment:
I think this options filter might not be necessary since we want these commands to be cloned 100%.